packages/jazz-tools/src/implementation/subscriptionScope.ts

```diff
@@ -41,30 +41,29 @@ export class SubscriptionScope<Root extends CoValue> {
     ) {
         this.rootEntry = {
             state: "loaded" as const,
-            value: root,
+            value: root._raw,
             rawUnsub: () => {}, // placeholder
         };
         this.entries.set(root.id, this.rootEntry);
 
         subscriptionsScopes.set(root, this);
 
         this.subscriber = root._loadedAs;
-        this.onUpdate = onUpdate;
+        this.scheduleUpdate = () => {
+            const value = rootSchema.fromRaw(this.rootEntry.value) as Root;
+            subscriptionsScopes.set(value, this);
+            onUpdate(value);
+        };
+
         this.rootEntry.rawUnsub = root._raw.core.subscribe(
             (rawUpdate: RawCoValue | undefined) => {
                 if (!rawUpdate) return;
-                this.rootEntry.value = rootSchema.fromRaw(rawUpdate) as Root;
-                // console.log("root update", this.rootEntry.value.toJSON());
-                subscriptionsScopes.set(this.rootEntry.value, this);
+                this.rootEntry.value = rawUpdate;
```

Comment:
@aeplay 

Managed to get the immutable updates working by moving the value view creation inside the `scheduleUpdate`

If this looks a good strategy for you I can proceed with some intensive testing to validate that I'm not breaking anything and there are no major performance regressions.